### PR TITLE
Feature/allow saving of models in cv new

### PR DESCRIPTION
**Pull request by mulhod**
Make a new `save_cv_models` field in the configuration file (alongside the similarly-named `save_cv_folds`) that allows one to save models trained for each fold during cross-validation.

Addesses #501.

### CODE REVIEW

**Comment by mulhod** [comment]
This PR is ready again. I hope that the coverage report is updated soon.

File: doc/run_experiment.rst [hunk 1]
```diff
@@ -1151,6 +1151,18 @@ Defaults to ``False``.
 
 .. _run_experiment:
 
+save_cv_models *(Optional)*
+""""""""""""""""""""""""""
```
Comment (desilinguist) [inline doc/run_experiment.rst]:
```suggestion
"""""""""""""""""""""""""""
```

I think you need to cover the full text and there seemed to be one missing.

File: doc/run_experiment.rst [hunk 2]
```diff
@@ -1151,6 +1151,18 @@ Defaults to ``False``.
...
+
+Whether to save the fold models that were generated during the
+cross-validation experiment to the :ref:`models` directory. Beware when using
+a grid engine that transferring lots of large model objects over the network,
+which this option would require, could be prolematic. Only set this option to
+``True`` if you understand the implications.
+Defaults to ``False``.
+
```
Comment (desilinguist) [inline doc/run_experiment.rst]: May be also say something about large amounts of storage before the grid thing?

**Comment by mulhod** [comment]
I don't really understand the code coverage report. In any case, I added in one test to cover a line that my PR introduces (testing whether `save_cv_models` is set to `true` and _not_ specifying a models output path will result in an invalid config file).

All the PR checks should now show up as completed.

Anybody have a moment for a review?

File: skll/experiments.py [hunk 3]
```diff
@@ -675,23 +679,29 @@ def _classify_featureset(args):
...
+                                              cv_folds=cv_folds,
+                                              grid_objective=grid_objective,
+                                              output_metrics=output_metrics,
+                                              param_grid=param_grid,
+                                              grid_jobs=grid_search_jobs,
+                                              save_cv_folds=save_cv_folds,
+                                              save_cv_models=save_cv_models,
+                                              use_custom_folds_for_grid_search=use_folds_file_for_grid_search)
+            if models:
+                for index, m in enumerate(models):
```
Comment (bndgyawali) [inline skll/experiments.py]: do we want the fold models index to start from 0 or 1?

Comment (mulhod) [reply skll/experiments.py]: I can switch it to 1 unless anybody has a strong preference for starting at 0? I guess the CV results start at fold 1.

Comment (desilinguist) [reply skll/experiments.py]: Yeah, I think matching the index to whatever the results shows makes sense.

Comment (mulhod) [reply skll/experiments.py]: I've made the change.

**Pull request merged**